cpp/ql/src/semmle/code/cpp/models/implementations/StdMap.qll

```diff
@@ -37,7 +41,8 @@ private class StdMapConstructor extends Constructor, TaintFunction {
  */
 private class StdMapInsert extends TaintFunction {
   StdMapInsert() {
-    this.hasQualifiedName("std", ["map", "unordered_map"], ["insert", "insert_or_assign"])
+    this.hasName(["insert", "insert_or_assign"]) and
+    this.getDeclaringType() instanceof MapOrUnorderedMap
```

Comment:
We should be using `this.getClassAndName(["insert", "insert_or_assign"]) instanceof MapOrUnorderedMap` in cases like this, as it was found to give a better join order (in fact, when the name was a `[]` set it made little difference, but consistency is good).

Reply:
Ah, it looks like this has been done since I opened up the diff.